src/plugins/vis_builder/public/application/components/data_tab/index.tsx

```diff
@@ -3,19 +3,336 @@
+
+        const destinationLimit = destinationSchema?.max;
+        if (destinationLimit && destinationAggFields.length <= destinationLimit) {
+          // Case 1: Destination has space
+          // We Need to update sourceAgg
+
+          createNewAggConfig({
+            fieldName: sourceFieldName,
+            sourceGroup: source.droppableId,
+            destinationSchema,
+            aggProps,
+            aggService,
+            sourceAgg,
+          });
+
+          // Remove the sourceAggConfig from the updated Config
+          const updatedAggConfig = aggProps.aggConfigs?.aggs.filter(
+            (agg) => agg.id !== sourceAggId
+          );
+
+          if (updatedAggConfig?.length) {
+            dispatch(updateAggConfigParams(updatedAggConfig.map((agg) => agg.serialize())));
+          }
+        } else {
+          // Case 2 : Destination has no space
+          return;
+        }
+      }
+    } else if (combine !== null) {
+      // Combining Elements
+      // TODO: Do we need to restrict drag and drop features amongst the Dragables in one Droppables
+
+      // Creating an Aggregation of the Source Field in the destination Schema
+      createNewAggConfig({
+        fieldName: sourceFieldName,
+        sourceGroup: source.droppableId,
+        destinationSchema,
+        aggProps,
+        aggService,
+        sourceAgg,
+      });
+
+      // Removing the previous destination and source AggId's
+      const updatedAggConfig = aggProps.aggConfigs?.aggs.filter(
+        (agg) => agg.id !== destinationAggId && agg.id !== sourceAggId
+      );
+
+      if (updatedAggConfig) {
+        dispatch(updateAggConfigParams(updatedAggConfig.map((agg) => agg.serialize())));
+      }
+    }
+  };
+
+  const handleDragEnd = ({ source, destination, combine, draggableId }: DropResult) => {
+    try {
+      const destinationSchemaName = destination?.droppableId || combine?.droppableId;
+      const sourceSchemaName = source.droppableId;
+
+      if (!sourceSchemaName || !destinationSchemaName) {
+        // Invalid Scenario source should be present
+        return;
+      }
+
+      const panelGroups = Array.from(schemas.all.map((schema) => schema.name));
+
+      // Transition from FieldSelector to Conifg panel
+      if (fieldSelectorGroups.includes(sourceSchemaName)) {
+        if (panelGroups.includes(destinationSchemaName)) {
+          handleFieldSelectorToConfigurationPanelTransition({
+            source,
+            destination,
+            combine,
+            draggableId,
+          } as DropResult);
+        }
+      } else if (panelGroups.includes(sourceSchemaName)) {
+        if (panelGroups.includes(destinationSchemaName)) {
+          handleConfigurationPanelTransition({
+            source,
+            destination,
+            combine,
+            draggableId,
+          } as DropResult);
+        }
+      }
+    } catch (err) {
+      return;
+    }
+  };
+
+  // What all state variables do we need
```

Comment:
These comments don't make sense to me.. 😆😆 😆  
You post a question then suggest that for "meta" fields, the component only requires the field names and no other state variables or properties? then talk about pane? 
Could you polish this comment a bit more?

Reply:
@ananzh I have cleaned up this comments. This comments were meant for me just so that I can remember the open questions that I had and where did I have those questions. I forgot to clean them up in the previous version.